### PR TITLE
VFS-111: Saving Module

### PR DESCRIPTION
**Pull request by 0xAtum**

### CODE REVIEW

File: src/main/SavingModule.sol [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+		totalVRR -= reward;
+		rewardAllocation += reward;
+
+		emit TotalVSTFromVRRUpdated(totalVRR);
+
+		if (reward != 0) {
+			emit PoolUpdated(reward, rewardAllocation);
+		}
+
+		if (minuteDifference == 0) return;
```
Comment (Paul4912) [inline src/main/SavingModule.sol:310]: can put this up earlier right after minute difference is created

File: src/main/SavingModule.sol [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+	}
+
+	function _refillReward() internal {
+		uint256 reward = _getRewardToRefill();
+		uint256 minuteDifference = (block.timestamp - lastUpdate) / 1 minutes;
+
+		totalVRR -= reward;
+		rewardAllocation += reward;
+
+		emit TotalVSTFromVRRUpdated(totalVRR);
```
Comment (Paul4912) [inline src/main/SavingModule.sol:304]: makes no functional difference but i think better to emit events at end to follow check-effects-interactions pattern

File: src/main/model/SavingModuleModel.sol [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity >=0.8.0;
+
+struct Lock {
+	//chunk
+	address user;
+	bool autoLock;
+	uint128 lockDay;
```
Comment (Paul4912) [inline src/main/model/SavingModuleModel.sol:8]: lockDay sounds like which day it was locked. i think you should add a s so its lockDays so its clearer this is number of days to lock it.

File: src/main/Shareable.sol [hunk 4]
```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { FullMath as Math } from "./lib/FullMath.sol";
+import { IShareable } from "./IShareable.sol";
+
+abstract contract Shareable is IShareable {
+	uint256 public share; // crops per gem    [ray]
```
Comment (Paul4912) [inline src/main/Shareable.sol:8]: was this taken from some farming nft game?

**Comment by Paul4912** [comment]
So if user wants to add to their deposit they have to make new lock essentially new deposit. Why not add feature to add to existing lock. Don't change anything just add amount so people can add to their "savings". Or would that ruin the way the accounting works. Seems doable to me.

File: src/main/Shareable.sol (hunk 4 above)
Comment (0xAtum) [reply src/main/Shareable.sol:8]: No it's from the CPD engine of Maker.

**Comment by 0xAtum** [comment]
> 

Because it goes against the Lock System & reward logic, for autolocking it would have been possible but it adds extra difficulty which wasn't worth to jump in

File: src/main/SavingModule.sol (hunk 1 above)
Comment (0xAtum) [reply src/main/SavingModule.sol:311]: I don't want to limit the refilling system with something that has nothing to do with the core logic

**Pull request merged**